docs: add documentation styleguide (#3007)
Sage/carbon: docs/documentation-styleguide.md

# Carbon Documentation Style Guide
## Follow these guidelines when contributing documentation to the Carbon repository

### Contents

[Introduction](#introduction)

[What We Document](#what-we-document)
* [User Facing Documentation](#user-facing-documentation)
* [Release Notes and Changelog](#release-notes-and-changelog)

> * [Request For Change (RFC)](#request-for-change)

Reviewer: Could you double check that this RFC skip-link works?

Author: I missed off "-rfc" 😞 I added that afterwards and didn't update the link. Corrected.

* [README files](#readme-files)
* [Setup and Getting Started Guides](#setup-and-getting-started-guides)
* [Styleguides](#styleguides)
* [Other Internal Documentation](#other-internal-documentation)
  * [New Joiners Guide](#new-joiners-guide)
  * [Carbon Roadmap](#carbon-roadmap)
  
[Language and Grammar Style](#language-and-grammar-style)
* [Jargon](#jargon)
* [Abbreviations (including Acronyms)](#abbreviations-including-acronyms)
* [Capitalisation](#capitalisation)

[Text Formatting and Organisation](#text-formatting-and-organisation)
* [Code in text](#code-in-text)
  * [Code string example](#code-string-example)
  * [Code block example](#code-block-example)
* [Bullet points and numbered lists](#bullet-points-and-numbered-lists)
* [Tables](#tables)

[Use of Links](#use-of-links)
* [Linking to other sites](#linking-to-other-sites)
* [Linking to other documents](#linking-to-other-documents)

[Use of Images](#use-of-images)

### Introduction

> This Documentation styleguide describes what is documented in the Carbon repository and which format this  takes to enable you to contribute documentation in line with the current styles. It includes:

Reviewer 2: ```suggestion
This Documentation style guide describes what is documented in the Carbon repository and which format this takes to enable you to contribute documentation in line with the current styles. It includes:
```

Author: I couldn't see that extra space between "this" and "takes" as the word "takes" had wrapped onto the next line in VS Code. Thanks 👍

Author: Separated "styleguide" into two words "style guide" throughout, apart from where it quotes another document.

- where documents live within the repository;
- which format is used for which purpose;
- language and grammar style;

> - linking to other documents and sites; and

Reviewer 2: ```suggestion
- linking to other documents and sites;
```

Author: I followed the first example in "A list of very short points" in this guide for bullet point lists: http://www.plainenglish.co.uk/punctuating-bulleted-lists.html. However, if you think the second example without an "and" before the last point, then I'm happy to change.

Author: Removed the "and" in all lists in this document.

#### Request For Change (RFC)
When you want to propose a substantial change to Carbon, such as the addition of a new component or a breaking change, you need to submit an RFC. The core Carbon Team will discuss the RFC needs to incorporate design constraints and reach a consensus before it is progressed.

There is a [template](../rfcs/template.md) to guide you when writing an RFC. You can add other sections if appropriate. Use hyphens to separate words in the file name and save it as a `.md` file in the `carbon/rfcs/text` directory. If you need to add supporting images, save them in the `carbon/rfcs/images` directory.

#### README files
We write README files in MD format to tell collaborators about what Carbon and [Cypress.io](https://www.cypress.io) are, how collaborators can get started and where they can find further information or assistance.

The [Carbon README](../README.md) contains:
- a link to a more detailed 'Getting Started' guide;

> - a link to the [contributing guide](../CONTRIBUTING.md) explaing how to contribute to the project; and

Reviewer 2: one more `and` missed :(

Author: 🤦‍♀️ Corrected.

- license information.

The [Cypress README](../cypress/README.md) contains:
- instructions of how to install and run Cypress;

> - how to access test results; and

Reviewer 2: ```suggestion
- how to access test results;
```

Author: I followed the first example in "A list of very short points" in this guide for bullet point lists: http://www.plainenglish.co.uk/punctuating-bulleted-lists.html. However, if you think the second example without an "and" before the last point, then I'm happy to change.

Author: Removed the "and" in all lists in this document.

- a link to the [testing styleguide](testing-styleguide.md) where a detailed explanation of how to write Cypress tests can be found.

The [RFCs README](../rfcs/README.md) provides detailed instructions of when and how to write an RFC. 

All README files are accessible through GitHub from their respective directories.

When writing a README file, consider:
- who your audience is;
- what useful information you want in there;
- if a table of contents is required;

> - the [use of links](#use-of-links); and

Reviewer 2: ```suggestion
- the [use of links](#use-of-links);
```

Author: I followed the first example in "A list of very short points" in this guide for bullet point lists: http://www.plainenglish.co.uk/punctuating-bulleted-lists.html. However, if you think the second example without an "and" before the last point, then I'm happy to change.

Author: Removed the "and" in all lists in this document.

#### Setup and Getting Started Guides
Similar to a README, these guides are written in MD or MDX format. They contain more detailed information about how collaborators set up their [development environment](dev-environment-setup.md) and [get started](getting-started.stories.mdx) with developing their application using Carbon. These documents are stored in the `/docs` directory and accessible from GitHub.

#### Styleguides
We write various styleguides to provide some consistency for collaborators to the Carbon project to follow. This helps to maintain a standard way of doing things so we have a limited set of methods, tools and styles adopted throughout the repository. They often contain ordered and concise instructions so users can follow and repeat them easily. This is particularly useful for people new to the project who seek some common guidelines.

Styleguides are stored in the `/docs` directory. It is a living directory so it continuously changes, with new guides added and others amended or removed. 

When writing a styleguide, consider:
- who your audience is and what prior understanding they have;
- if a glossary is required;
- formatting of [code in text](#code-in-text);

> - the [use of links](#use-of-links); and

Reviewer 2: ```suggestion
- the [use of links](#use-of-links);
```

Author: I followed the first example in "A list of very short points" in this guide for bullet point lists: http://www.plainenglish.co.uk/punctuating-bulleted-lists.html. However, if you think the second example without an "and" before the last point, then I'm happy to change.

Author: Removed the "and" in all lists in this document.
